File: 1122 - Relative Sort Array.java
class Solution {
    public int[] relativeSortArray(int[] arr1, int[] arr2) {
        int[] f = new int[1001], res = new int[arr1.length];
        for (int n : arr1) {
            f[n]++;
        }
        int idx = 0;
        for (int n : arr2) {
            for (int i = 0; i < f[n]; i++) {
                res[idx++] = n;
            }
            f[n] = 0;
        }
        for (int i = 0; i < 1001; i++) {
            for (int j = 0; j < f[i]; j++) {
                res[idx++] = i;
            }
        }
        return res;
    }
}

File: 1122 - Relative Sort Array.php
<?php

class Solution {


/**
* @param Integer[] $arr1
* @param Integer[] $arr2
* @return Integer[]
*/
function relativeSortArray($arr1, $arr2) {
    $f = array_fill(0, 1001, 0);
    $res = array_fill(0, count($arr1), 0);
    $idx = 0;
    foreach ($arr1 as $n) {
        $f[$n]++;
    }
    foreach ($arr2 as $n) {
        for ($i = 0; $i < $f[$n]; $i++) {
            $res[$idx++] = $n;
        }
        $f[$n] = 0;
    }
    for ($i = 0; $i < 1001; $i++) {
        for ($j = 0; $j < $f[$i]; $j++) {
            $res[$idx++] = $i;
        }
    }
    return $res;
}
}